share/genbuild.sh

```diff
@@ -19,7 +19,11 @@ if [ -e "$(which git)" ]; then
     DESC="$(git describe --dirty 2>/dev/null)"
 
     # get only commit hash, but format like describe
-    DESCHASH="$(git log -n 1 --format="%h")"
+    DESCHASH="$(git rev-parse --short=9 HEAD)"
```

Comment:
We should probably send errors to null or this will produce warnings when running on a downloaded zip archive.

```
marco@dev:~/Downloads/Gridcoin-Research-master$ sh ./share/genbuild.sh apa.h
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```